### DISCUSSION TITLE
TSML UI does not work when templates are placed in theme directory

### DISCUSSION THREAD

**Discussion by imramugh** [code4recovery/12-step-meeting-list#990] [General]
I have a brand new install at:  https://aatorontoprdev.wpengine.com

In the meeting settings, I've selected TSML UI.  This works fine, however in order to get it to work with my theme, when I place the templates in my theme's root folder, it kicks back to the legacy UI.  Is this the expected behaviour?

I'm using Theme.co's X Pro theme.

My desired end-state is to use the TSML UI (which I think looks great), within my current theme.  Any help you can provide would be greatly appreciated.

Thanks!

**Comment by joshreisner** [comment]
Hey Imran, great to see you're still involved with GTA this many years later.

I think we might have a little bug here. We now have two template files, `archive-meetings.php` (Legacy UI) and `archive-tsml-ui.php` (TSML UI). 

Looks like if you bring the legacy file to your theme you will see it enabled, but if you bring the TSML UI file to your theme it doesn't pick it up. I can create an issue to fix that. 

For now, you might just copy `archive-tsml-ui.php` to your theme and then rename it to `archive-meetings.php`

**Comment by imramugh** [comment]
Thanks for the prompt response, Josh – as always.  Yeah, I’m coming back to GTA intergroup after a 3-year hiatus (in the spirit of rotation).  Covid didn’t help one bit… our meeting list is a complete train-wreck, so while I’m fixing the data I figured now would be a good time to clean up the site too and get a bit more modern, hence the request.  Looks like 90% of our traffic is mobile so my priority is to re-do the site to be mobile-first.

Thank you again.  I’ll give it a go with the archive-tsml-ui.php file and see how that works.

Thanks!

IM.

--
Imran Mughal
(416) 846-5561

From: Josh Reisner ***@***.***>
Date: Thursday, December 8, 2022 at 10:21 AM
To: code4recovery/12-step-meeting-list ***@***.***>
Cc: Imran M. ***@***.***>, Author ***@***.***>
Subject: Re: [code4recovery/12-step-meeting-list] TSML UI does not work when templates are placed in theme directory (Discussion #990)

Hey Imran, great to see you're still involved with GTA this many years later.

I think we might have a little bug here. We now have two template files, archive-meetings.php (Legacy UI) and archive-tsml-ui.php (TSML UI).

Looks like if you bring the legacy file to your theme you will see it enabled, but if you bring the TSML UI file to your theme it doesn't pick it up. I can create an issue to fix that.

For now, you might just copy archive-tsml-ui.php to your theme and then rename it to archive-meetings.php

**Comment by imramugh** [comment]
Renaming the file and putting it in the theme directory worked.

That said, previously I remember the file being way more involved and I was able to update the column span, etc. to make it fit within the theme.  This time around the page seems to be very barebones.  I was hoping that putting it in my theme directory would mean that it would respect the default page template/container.  Is there any documentation I can look at?  I don’t want to trouble you for every little thing.  😊

IM.

--
Imran Mughal
(416) 846-5561

From: Imran Mughal ***@***.***>
Date: Thursday, December 8, 2022 at 10:24 AM
To: code4recovery/12-step-meeting-list ***@***.***>, code4recovery/12-step-meeting-list ***@***.***>
Cc: Author ***@***.***>
Subject: Re: [code4recovery/12-step-meeting-list] TSML UI does not work when templates are placed in theme directory (Discussion #990)
Thanks for the prompt response, Josh – as always.  Yeah, I’m coming back to GTA intergroup after a 3-year hiatus (in the spirit of rotation).  Covid didn’t help one bit… our meeting list is a complete train-wreck, so while I’m fixing the data I figured now would be a good time to clean up the site too and get a bit more modern, hence the request.  Looks like 90% of our traffic is mobile so my priority is to re-do the site to be mobile-first.

Thank you again.  I’ll give it a go with the archive-tsml-ui.php file and see how that works.

Thanks!

IM.

--
Imran Mughal
(416) 846-5561

From: Josh Reisner ***@***.***>
Date: Thursday, December 8, 2022 at 10:21 AM
To: code4recovery/12-step-meeting-list ***@***.***>
Cc: Imran M. ***@***.***>, Author ***@***.***>
Subject: Re: [code4recovery/12-step-meeting-list] TSML UI does not work when templates are placed in theme directory (Discussion #990)

Hey Imran, great to see you're still involved with GTA this many years later.

I think we might have a little bug here. We now have two template files, archive-meetings.php (Legacy UI) and archive-tsml-ui.php (TSML UI).

Looks like if you bring the legacy file to your theme you will see it enabled, but if you bring the TSML UI file to your theme it doesn't pick it up. I can create an issue to fix that.

For now, you might just copy archive-tsml-ui.php to your theme and then rename it to archive-meetings.php

**Reply by joshreisner** [reply]
that type of change would be a CSS change, for example:

```
#tsml-ui { max-width: 1400px; }
```

i see your theme is adding a few extra CSS rules to the controls row that we're not used to seeing. try adding:

```
#tsml-ui form { margin: 0; }
#tsml-ui form input { height: auto; }
```

many themes have the ability to add "Additional CSS" under "Appearance > Customize" - you might not need to override

**Reply by joshreisner** [reply]
actually you can disregard that second part - i will add some stuff to TSML to prevent that

**Comment by imramugh** [comment]
Thanks Josh… you da man!!


--
Imran Mughal
(416) 846-5561

From: Josh Reisner ***@***.***>
Date: Thursday, December 8, 2022 at 11:59 AM
To: code4recovery/12-step-meeting-list ***@***.***>
Cc: Imran M. ***@***.***>, Author ***@***.***>
Subject: Re: [code4recovery/12-step-meeting-list] TSML UI does not work when templates are placed in theme directory (Discussion #990)

actually you can disregard that second part - i will add some stuff to TSML to prevent that

**Comment by imramugh** [comment]
There’s some weirdness going on with stylesheets when using the TSML UI.

Background:  I’ve built a number of sites using Theme.co’s X Pro – so I’m quite familiar with it.  My expectation is that I can put in additional CSS either in the child theme’s style.css, or through the front-end.  However, when I do, nothing seems to happen.  However, when I use wordrpess’ customize > css function, it behaves and includes that css.

[Graphical user interface, text, application  Description automatically generated]

However, anywhere else… and it does not work.  Here’s a screenshot of the theme’s global.css
[Graphical user interface  Description automatically generated]

And another one of me putting it in the child theme’s style.css
 [Graphical user interface, text  Description automatically generated]


Lastly, when using the TSML UI, I get double scroll bars when testing for mobile devices:
[Graphical user interface, application  Description automatically generated]


Hopefully, some of this stuff is easy and just a case of me being unfamliar with the new interface since I’m coming back to this role after 3 years.  😊

Any help you can provide would be greatly appreicated.

Thanks Josh.

IM.

--
Imran Mughal
(416) 846-5561

From: Josh Reisner ***@***.***>
Date: Thursday, December 8, 2022 at 11:59 AM
To: code4recovery/12-step-meeting-list ***@***.***>
Cc: Imran M. ***@***.***>, Author ***@***.***>
Subject: Re: [code4recovery/12-step-meeting-list] TSML UI does not work when templates are placed in theme directory (Discussion #990)

actually you can disregard that second part - i will add some stuff to TSML to prevent that

**Reply by joshreisner** [reply]
hey imran, i'm not seeing any of those screenshots

i'm also not seeing double scroll on mobile, but the `width: 88%` is not great on mobile sizes, maybe remove or put inside a media query?

i think TSML UI will work better if you can change your permalink structure to not use the default query string method

**Comment by imramugh** [comment]
And as usual, here’s a link to the site:  https://aatorontoprdev.wpengine.com

--
Imran Mughal
(416) 846-5561

From: Imran Mughal ***@***.***>
Date: Thursday, December 8, 2022 at 8:56 PM
To: code4recovery/12-step-meeting-list ***@***.***>, code4recovery/12-step-meeting-list ***@***.***>
Cc: Author ***@***.***>
Subject: Re: [code4recovery/12-step-meeting-list] TSML UI does not work when templates are placed in theme directory (Discussion #990)
There’s some weirdness going on with stylesheets when using the TSML UI.

Background:  I’ve built a number of sites using Theme.co’s X Pro – so I’m quite familiar with it.  My expectation is that I can put in additional CSS either in the child theme’s style.css, or through the front-end.  However, when I do, nothing seems to happen.  However, when I use wordrpess’ customize > css function, it behaves and includes that css.

[Graphical user interface, text, application  Description automatically generated]

However, anywhere else… and it does not work.  Here’s a screenshot of the theme’s global.css
[Graphical user interface  Description automatically generated]

And another one of me putting it in the child theme’s style.css
 [Graphical user interface, text  Description automatically generated]


Lastly, when using the TSML UI, I get double scroll bars when testing for mobile devices:
[Graphical user interface, application  Description automatically generated]


Hopefully, some of this stuff is easy and just a case of me being unfamliar with the new interface since I’m coming back to this role after 3 years.  😊

Any help you can provide would be greatly appreicated.

Thanks Josh.

IM.

--
Imran Mughal
(416) 846-5561

From: Josh Reisner ***@***.***>
Date: Thursday, December 8, 2022 at 11:59 AM
To: code4recovery/12-step-meeting-list ***@***.***>
Cc: Imran M. ***@***.***>, Author ***@***.***>
Subject: Re: [code4recovery/12-step-meeting-list] TSML UI does not work when templates are placed in theme directory (Discussion #990)

actually you can disregard that second part - i will add some stuff to TSML to prevent that

**Comment by imramugh** [comment]
Yeah the original email bounced b/c of the attachment size.  Lemme try again.

Correct… I normally use a media query.  For mobile I’ll typically go 100% since real estate is at a premium.

[Graphical user interface, text, application  Description automatically generated][Graphical user interface  Description automatically generated][Graphical user interface, text  Description automatically generated][Graphical user interface, application  Description automatically generated]


--
Imran Mughal
(416) 846-5561

From: Josh Reisner ***@***.***>
Date: Thursday, December 8, 2022 at 9:07 PM
To: code4recovery/12-step-meeting-list ***@***.***>
Cc: Imran M. ***@***.***>, Author ***@***.***>
Subject: Re: [code4recovery/12-step-meeting-list] TSML UI does not work when templates are placed in theme directory (Discussion #990)

hey imran, i'm not seeing any of those screenshots

i'm also not seeing double scroll on mobile, but the width: 88% is not great on mobile sizes, maybe remove or put inside a media query?

i think TSML UI will work better if you can change your permalink structure to not use the default query string method

**Reply by joshreisner** [reply]
i don't think email attachments are rendered. i think you need to use the website

**Comment by imramugh** [comment]
BTW, remember how with the old UI we could make a copy of the php templates, and then make changes to these without worrying about them being overwritten when an update was done?  I’m wondering if there’s a similar capability with the new TSML UI to allow for styling changes?

Specifically, I don’t want to use the default fonts or colors.  I want to be able to override those to match the rest of my wordpress theme.

I’ve got a working prototype of our new site here:  https://aatorontoprdev.wpengine.com/meetings/ as a reference.

Thanks!

IM.

--
Imran Mughal
(416) 846-5561

From: Josh Reisner ***@***.***>
Date: Thursday, December 8, 2022 at 9:41 PM
To: code4recovery/12-step-meeting-list ***@***.***>
Cc: Imran M. ***@***.***>, Author ***@***.***>
Subject: Re: [code4recovery/12-step-meeting-list] TSML UI does not work when templates are placed in theme directory (Discussion #990)

i don't think email attachments are rendered. i think you need to use the website

**Reply by joshreisner** [reply]
hey, looks good! 

you can update fonts and colors using CSS 👍 

example:
```
#tsml-ui div.tsml-ui {
  font-family: Georgia !important;
}
#tsml-ui div.tsml-ui a {
  color: hotpink;
}
#tsml-ui div.tsml-ui a.online {
  background-color: hotpink;
  border: 0;
  color: white !important;
}
#tsml-ui div.tsml-ui a.in-person {
  background-color: plum;
  border: 0;
  color: purple !important;
}
```

**Comment by imramugh** [comment]
Is it possible to cancel a single instance of a meeting?  Asking because I just got Informed by the church that the basement will not be available on a particular date for a church event.  Can I go in and put that date in so that the meeting is suppressed on that date?IM—Imran Mughal(416) 846-5561On Jan 5, 2023, at 6:30 PM, Josh Reisner ***@***.***> wrote:﻿
hey, looks good!
you can update fonts and colors using CSS 👍
example:
#tsml-ui div.tsml-ui {
  font-family: Georgia !important;
}
#tsml-ui div.tsml-ui a {
  color: hotpink;
}
#tsml-ui div.tsml-ui a.online {
  background-color: hotpink;
  border: 0;
  color: white !important;
}
#tsml-ui div.tsml-ui a.in-person {
  background-color: plum;
  border: 0;
  color: purple !important;
}

**Reply by tim-rohrer** [reply]
There really isn't a (good) way to do this for a single event like you're describing. You *could*, in theory, put the meeting in draft after the last valid session, and then republish it after the date the meeting will not occur. But, frankly, that is more work than is probably necessary and most people probably won't notice.

Likely best to add in notes or announce through your website.

**Reply by joshreisner** [reply]
i would set it to “people cannot attend in person” after the last valid session and set it back after. this will make the meeting show up as inactive / location temporarily closed on the meeting list and in the app

**Comment by imramugh** [comment]
Yeah, there really isn’t an ideal way to deal with it.  However, we’re lucky that it’s a weekly meeting, so we have time to update the listing – so that’s what we’ll do.  I’ll also update the meeting notes with a blurb so that if someone was to look it up the week prior, they would know that it’s not on the following week.

Just thought I’d check to see if there was anything else you might have up your sleeve.

Thanks again for your prompt response, as always.  Really appreciate the work that you do to help keep the wheels of AA turning.  😊

IM.

--
Imran Mughal
(416) 846-5561

From: Josh Reisner ***@***.***>
Date: Thursday, February 16, 2023 at 7:54 PM
To: code4recovery/12-step-meeting-list ***@***.***>
Cc: Imran M. ***@***.***>, Author ***@***.***>
Subject: Re: [code4recovery/12-step-meeting-list] TSML UI does not work when templates are placed in theme directory (Discussion #990)

i would set it to “people cannot attend in person” after the last valid session and set it back after. this will make me meeting show up as inactive / location temporarily closed on the meeting list and in the app
